dev/io.openliberty.microprofile.openapi.2.0.internal/src/io/openliberty/microprofile/openapi20/merge/RenameReferenceVisitor.java

```diff
@@ -0,0 +1,230 @@
+        }
+        return link;
+    }
+
+    private static final Pattern PATH_REF_PATTERN = Pattern.compile("#/paths/([^/]+)/.*");
+
+    /**
+     * References to operations include a path as part of the reference which may have been renamed as part of the merge process.
+     * <p>
+     * This method updates the operation reference to use the new path if necessary
+     * 
+     * @param operationRef the operation reference, may be {@code null}
+     * @return the updated operation reference
+     */
+    private String updateOperationRef(String operationRef) {
+        if (operationRef == null) {
+            return null;
+        }
+
+        Matcher m = PATH_REF_PATTERN.matcher(operationRef);
+        if (!m.matches()) {
+            return operationRef;
+        }
+
+        String oldPath = decodePathReference(m.group(1));
+        String newPath = renameHolder.lookupName(NameType.PATHS, oldPath);
+
+        if (!oldPath.equals(newPath)) {
+            String newRef = operationRef.substring(0, m.start(1))
+                            + encodePathReference(newPath)
+                            + operationRef.substring(m.end(1));
+            return newRef;
+        } else {
+            return operationRef;
+        }
+
+    }
+
+    private String decodePathReference(String pathReference) {
+        return pathReference.replace("~1", "/").replace("~0", "~");
+    }
+
+    private String encodePathReference(String path) {
+        return path.replace("~", "~0").replace("/", "~1");
+    }
+
+    @Override
+    public Parameter visitParameter(Context context, Parameter p) {
+        updateReference(p, NameType.PARAMETERS, "#/components/parameters/");
+        return p;
+    }
+
+    @Override
+    public Parameter visitParameter(Context context, String key, Parameter p) {
+        updateReference(p, NameType.PARAMETERS, "#/components/parameters/");
+        return p;
+    }
+
+    @Override
+    public RequestBody visitRequestBody(Context context, RequestBody rb) {
+        updateReference(rb, NameType.REQUEST_BODIES, "#/components/requestBodies/");
+        return rb;
+    }
+
+    @Override
+    public RequestBody visitRequestBody(Context context, String key, RequestBody rb) {
+        updateReference(rb, NameType.REQUEST_BODIES, "#/components/requestBodies/");
+        return rb;
+    }
+
+    @Override
+    public APIResponse visitResponse(Context context, String key, APIResponse response) {
+        updateReference(response, NameType.RESPONSES, "#/components/responses/");
+        return response;
+    }
+
+    @Override
+    public Schema visitSchema(Context context, Schema schema) {
+        updateReference(schema, NameType.SCHEMAS, "#/components/schemas/");
+        return schema;
+    }
+
+    @Override
+    public Schema visitSchema(Context context, String key, Schema schema) {
+        updateReference(schema, NameType.SCHEMAS, "#/components/schemas/");
```

Comment:
This method has the same impl as the method on 177

Reply:
I've made this change. I'm not entirely sure about it, but it does at least reduce the duplication of code slightly and reinforces that the implementation of both methods is the same.

I'm not totally sure whether this is worth adding the extra step of redirection or calling a method declared on the interface in a different way to how it was designed. However, we did spend way too long thinking about it.